### PR TITLE
Epu refactoring

### PR DESCRIPTION
**Pull request by rbatistacardoso**

### CODE REVIEW

File: si-id-epu50/si_id_epu50/ecodrive.py [hunk 1]
```diff
@@ -2,96 +2,87 @@
...
-
-    def __init__(self, address, max_limit=+25, min_limit=-25,
-                    bbb_hostname = None, rs458_tcp_port=None, drive_name = 'EcoDrive') -> None:
-        """."""
+    _TCP_SOCKET_TIMEOUT = 1 # ms
+
+    def __init__(self, socket: socket.socket, address: int, max_limit, min_limit,
+                 bbb_hostname, rs458_tcp_port: int = None, drive_name: str = 'EcoDrive') -> None:
+        
+        """Intramat EcoDirve 03 (controllers DKC**.3-040, -100, -200) class.
```
Comment (xresende) [inline si-id-epu50/si_id_epu50/ecodrive.py:20]: typo

File: si-id-epu50/si_id_epu50/ecodrive.py [hunk 2]
```diff
@@ -2,96 +2,87 @@
...
         # self.set_rs485_delay(1)
 
     def tcp_wait_connection(self) -> bool:
-        """TCP wait connection.
-
+        """
         Any time this function is called, it keeps on a loop trying to reach
-        the other side of a tcp connection, when it succeed,
-        it returnts True.
+        the other side of a tcp connection, when it succeed, it returnts True.
```
Comment (xresende) [inline si-id-epu50/si_id_epu50/ecodrive.py:48]: typo

File: si-id-epu50/si_id_epu50/epu.py [hunk 3]
```diff
@@ -1,63 +1,63 @@
...
-import logging.handlers as handlers
-import threading, socket
-from datetime import datetime
+import logging.handlers
+import time
 
 from . import constants as _cte
 from .utils import *
-from .ecodrive import EcoDrive
+from .ecodrive import *
```
Comment (xresende) [inline si-id-epu50/si_id_epu50/epu.py:9]: is it possible to import only referenced symbols? we find it hard to keep track of symbols with wildcard imports.

File: si-id-epu50/si_id_epu50/ecodrive.py (hunk 1 above)
Comment (rbatistacardoso) [reply si-id-epu50/si_id_epu50/ecodrive.py]: Fixed

File: si-id-epu50/si_id_epu50/ecodrive.py (hunk 2 above)
Comment (rbatistacardoso) [reply si-id-epu50/si_id_epu50/ecodrive.py]: Fixed

File: si-id-epu50/si_id_epu50/epu.py (hunk 3 above)
Comment (rbatistacardoso) [reply si-id-epu50/si_id_epu50/epu.py]: Changed to `from .ecodrive import EcoDrive`

File: si-id-epu50/scripts/utils_logging.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+'''
```
Comment (anacso17) [inline si-id-epu50/scripts/utils_logging.py:1]: I think functions defined here could be in si_id_epu50 package

**Review by anacso17 — CHANGES_REQUESTED** [review]
Maybe we could also remove Dockerfile and docker-compose.yaml files in this PR

File: si-id-epu50/scripts/utils_logging.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+https://www.toptal.com/python/in-depth-python-logging
+'''
+import logging
+import logging.handlers
```
Comment (fernandohds564) [inline si-id-epu50/scripts/utils_logging.py:5]: this import is not necessary, is it?

File: si-id-epu50/scripts/utils_logging.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    fh = logging.handlers.RotatingFileHandler(file, maxBytes=1000000, backupCount=10)
+    fh.setLevel(logging.INFO)
+    fh.setFormatter(logging.Formatter("%(asctime)s — %(name)s — %(levelname)s — %(funcName)s:%(lineno)d — %(message)s"))
+    return fh
+
+def get_logger(name, file_handler):
+    lg = logging.getLogger(name)
+    lg.setLevel(logging.INFO)
+    lg.addHandler(file_handler)
+    return lg
```
Comment (fernandohds564) [inline si-id-epu50/scripts/utils_logging.py:21]: According to my linter there should be an empty line at the end of the file here.

File: si-id-epu50/si_id_epu50/ecodrive.py [hunk 7]
```diff
@@ -2,96 +2,87 @@
 
 import logging
 logger = logging.getLogger(__name__)
-import logging.handlers as handlers
 import socket
 import threading
 import time
 
 from . import constants as _cte
 from .utils import *
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/ecodrive.py:10]: try not to import this way. Maybe `from . import utils` or `import .utils as _utils` would be better.

File: si-id-epu50/si_id_epu50/ecodrive.py [hunk 8]
```diff
@@ -2,96 +2,87 @@
...
-    _SOCKET_TIMEOUT = 1 # tcp socket timeout
-
-    def __init__(self, address, max_limit=+25, min_limit=-25,
-                    bbb_hostname = None, rs458_tcp_port=None, drive_name = 'EcoDrive') -> None:
-        """."""
+    _TCP_SOCKET_TIMEOUT = 1 # ms
+
+    def __init__(self, socket: socket.socket, address: int, max_limit, min_limit,
+                 bbb_hostname, rs458_tcp_port: int = None, drive_name: str = 'EcoDrive') -> None:
+
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/ecodrive.py:19]: There should no be an empty line before the method docstring. And it is necessary to include an empty line between the first docstring line and the other ones.

File: si-id-epu50/si_id_epu50/ecodrive.py [hunk 9]
```diff
@@ -2,96 +2,87 @@
...
+        
+        """Indramat EcoDirve 03 (controllers DKC**.3-040, -100, -200) class.
+           Encaptulate ASCII messages into TCP datagrams.
+           Makes available a subset of possible readings and
+           writes from/to EcoDrive 03 controllers. It is required that on the
+           other side of communication, there is some software
+           unwrapping tcp messages and delivering it to a RS485 interface; on CNPEM
+           this is typically done with a beagle bone black configured properly.
+        """
+
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/ecodrive.py:28]: there is no emptly line after the docstring too.

File: si-id-epu50/si_id_epu50/epu.py [hunk 10]
```diff
@@ -1,63 +1,63 @@
+import threading
+import socket
 import logging
-logger = logging.getLogger(__name__)
-import logging.handlers as handlers
-import threading, socket
-from datetime import datetime
+import logging.handlers
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/epu.py:4]: this import is not necessary.

File: si-id-epu50/si_id_epu50/epu.py [hunk 11]
```diff
@@ -1,63 +1,63 @@
...
+import time
 
 from . import constants as _cte
 from .utils import *
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/epu.py:8]: try to avoid importing modules this way.

File: si-id-epu50/si_id_epu50/epu.py [hunk 12]
```diff
@@ -1,63 +1,63 @@
...
-logHandler = handlers.RotatingFileHandler(filename='epu.log', maxBytes=10*1024*1024)
-logHandler.setFormatter(formatter)
-logger.addHandler(logHandler)
-logger.info(datetime.now().strftime('%m/%d/%Y, %H:%M:%S'))
+logger = logging.getLogger(__name__)
+class Namespace:
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
 
+default_args = Namespace(pv_prefix='SI-10SB:ID-EPU50:', msg_port=5052, io_port=5050, beaglebone_addr='10.128.110.160')
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/epu.py:17]: there should be two lines separating classes definitions.

File: si-id-epu50/si_id_epu50/epu.py [hunk 13]
```diff
@@ -1143,3 +1144,60 @@ def turn_on_all(self):
...
         self.phase_turn_on()
+
+
+def get_file_handler(file: str):
+    # logger.handlers.clear()
+    fh = logging.handlers.RotatingFileHandler(file, maxBytes=10000000, backupCount=10)
+    fh.setLevel(logging.DEBUG)
+    fh.setFormatter(logging.Formatter("%(asctime)s — %(name)s — %(levelname)s — %(funcName)s:%(lineno)d — %(message)s"))
+    return fh
+
```
Comment (fernandohds564) [inline si-id-epu50/si_id_epu50/epu.py:1155]: there should be two lines separating function definitions.

**Review by fernandohds564 — CHANGES_REQUESTED** [review]
hi @rbatistacardoso, i noticed several linting issues with this PR. Which linter do you use there to check these problems? I think we use pylama and pylint here.

Could you take a look at these standardization issues, please?

**Comment by anacso17** [comment]
@rbatistacardoso I pushed a commit implementing some linting suggestions and updating the version file :)

File: si-id-epu50/si_id_epu50/connection_handler.py [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+                    else:
+                        data += chunk_str
+
+                return data
+
+        except ConnectionResetError:
+            logger.error("Connection lost. Retrying...")
+            self.connected = False
+            self.connect()
+            return self.receive_data('utf-8', errors='ignore')
```
Comment (anacso17) [inline si-id-epu50/si_id_epu50/connection_handler.py:103]: It seems to be a bug: there is no keyword `errors` in `receive_data` method

**Review by fernandohds564 — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Pull request merged**